R/ops-addition.r

```diff
@@ -16,8 +16,16 @@ add_duration_to_date <- function(dur, date) {
   if(is.Date(date)){
     date <- as.POSIXct(date)
     ans <- with_tz(date + dur@.Data, "UTC")
-    if (hour(ans) == 0 && minute(ans) == 0 && second(ans) == 0)
+
+    if ( all(is.na(ans) ) ) return(as.Date(ans))  # ALL NAs
+
+    if ( all( hour(na.omit(ans)) == 0 ) &&
```

Comment:
These 4 `all()` are not needed. Simply doing `all(hour(ans) == 0 & minute(ans) == 0 & second(ans) == 0, na.rm = T)` within that `if` should do the job.